core/src/main/java/org/wso2/msf4j/internal/MSF4JMessageProcessor.java

```diff
@@ -167,7 +256,275 @@ public void setTransportSender(TransportSender transportSender) {
+                                parameterList.add(closeReason);
+                            } else if (parameter.getType() == Session.class) {
+                                parameterList.add(session);
+                            } else if (parameter.getType() == String.class) {
+                                PathParam pathParam = parameter.getAnnotation(PathParam.class);
+                                if (pathParam != null) {
+                                    parameterList.add(paramValues.get(pathParam.value()));
+                                } else {
+                                    throw new IllegalArgumentException("String parameters without" +
+                                                                                     " @PathParam annotation");
+                                }
+                            } else {
+                                parameterList.add(null);
+                            }
+                        }
+                );
+                executeMethod(method, webSocketEndpoint, parameterList, session);
+            }
+        } catch (Throwable throwable) {
+            handleError(closeCarbonMessage, throwable, routableEndpoint, session);
+        }
+    }
+
+    /*
+    handle Control Carbon Message.
+    This is mapped to PongMessage in javax.websocket
+     */
+    private void handleControlCarbonMessage(ControlCarbonMessage controlCarbonMessage, PatternPathRouter.
+            RoutableDestination<Object> routableEndpoint, Session session) {
+        Object webSocketEndpoint = routableEndpoint.getDestination();
+        Map<String, String> paramValues = routableEndpoint.getGroupNameValues();
+        Method method = new EndpointDispatcher().getOnPongMessageMethod(webSocketEndpoint);
+        if (method != null) {
+            List<Object> parameterList = new LinkedList<>();
+            Arrays.stream(method.getParameters()).forEach(
+                    parameter -> {
+                        if (parameter.getType() == PongMessage.class) {
+                            parameterList.add(new WebSocketPongMessage(controlCarbonMessage.readBytes()));
+                        } else if (parameter.getType() == Session.class) {
+                            parameterList.add(session);
+                        } else if (parameter.getType() == String.class) {
+                            PathParam pathParam = parameter.getAnnotation(PathParam.class);
+                            if (pathParam != null) {
+                                parameterList.add(paramValues.get(pathParam.value()));
+                            } else {
+                                throw new IllegalArgumentException("String parameters " +
+                                                                           "without @PathParam annotation");
+                            }
+                        } else {
+                            parameterList.add(null);
```

Comment:
IMO shouldn't we add new Object() here to avoid possible null pointer exceptions. Consider using new Object()